daisy_workflows/linux_common/utils/guestfsprocess.py

```diff
@@ -23,26 +23,31 @@
 containing stdout, stderr, and the return code of the process.
 """
 
+import logging
 import os
 import shlex
 import textwrap
 import typing
 
 
-def run(g: 'GuestFSInterface', command) -> 'CompletedProcess':
+def run(g: 'GuestFSInterface', command,
+        check: bool = True) -> 'CompletedProcess':
   """Runs a process in a mounted GuestFS instance, ensuring that
   standard output and standard error is always retained.
 
   Args:
     g: Mounted GuestFS instance.
     command (str or List[str]): Script content that will be executed
     by a bash interpeter on the guest.
+    check (bool): When true and the process exits with a non-zero exit code,
```

Comment:
The name of this flag doesn't seem appropriate. It sounds like it's an instruction to check something but it's actually printing more debugging info and throwing an exception.

Reply:
I've modeled this after the standard library so that it'd be familiar to future programmers: https://docs.python.org/3/library/subprocess.html#subprocess.run

That said, any thoughts on something that would be clearer?

Reply:
I saw the reference to subprocess.run, but I still find it unintuitive. What does check mean there?

Reply:
My guess is: "check the result". I could prepend this to the docstring, something like: "Whether to check the result of the command. When true..."

Reply:
Sounds to me more like `raiseOnError` than `check`

Reply:
sgtm -- changed to `raiseOnError`